azure-pipelines.yml

```diff
@@ -80,16 +96,21 @@ steps:
     addChangeLog: false
 
 - task: Bash@3
-  condition: and(succeeded(), startsWith(variables['Build.SourceBranch'], 'refs/tags/'))
-  displayName: "nuget Publishing"
+  condition: and(succeeded(), eq(variables['NUGET_PUBLISH_FLAG'], 'true'))
   inputs:
     targetType: 'inline'
     script: |
-      echo $(Build.SourceBranch) 
-      if [[ $(Build.SourceBranch) =~ ^refs/tags/v?[0-9]+\.[0-9]+\.[0-9]+$ ]]; then
-          echo "starting NuGet Push"
-          nuget push *.nupkg $(apikey) -Source https://api.nuget.org/v3/index.json
-      else
-          echo "skipping NuGet Push"
-      fi
-    workingDirectory: '$(Build.ArtifactStagingDirectory)'
\ No newline at end of file
+      wget https://aka.ms/downloadazcopy-v10-linux
+      tar -xvf downloadazcopy-v10-linux
+      sudo cp ./azcopy_linux_amd64_*/azcopy /usr/bin/
+
+- task: Bash@3
+  displayName: 'Uplaod to Blob'
+  condition: and(succeeded(), eq(variables['NUGET_PUBLISH_FLAG'], 'true'))
+  inputs:
+    targetType: 'inline'
+    script: |
+      export AZCOPY_SPA_CLIENT_SECRET=$(AZURESDKPARTNERDROPS-SERVICE-PRINCIPAL-KEY)
+      azcopy login --service-principal --application-id $(APPID)
+      azcopy copy "$(Build.ArtifactStagingDirectory)/Microsoft.Azure.IoT.Edge.Module.$(PACKAGEVERSION).nupkg"  "https://azuresdkpartnerdrops.blob.core.windows.net/drops/azure-iot-edge/dotnet/$(PACKAGEVERSION)/Microsoft.Azure.IoT.Edge.Module.$(PACKAGEVERSION).nupkg"
```

Comment:
I think we need one more directory between 'dotnet' and '$(PACKAGEVERSION) - something unique to identify the package.